experiments/compositions/composition/internal/controller/expander_reconciler.go

```diff
@@ -324,6 +335,16 @@ func (r *ExpanderReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		newStatus.ClearCondition(compositionv1alpha1.Ready)
 		message := fmt.Sprintf("Evaluated stages: %s \n Applied stages: %s", strings.Join(stagesEvaluated, ", "), strings.Join(stagesApplied, ", "))
 		newStatus.AppendCondition(compositionv1alpha1.Ready, metav1.ConditionFalse, message, "PendingStages")
+
+		if expanderDebugLogsEnabled {
+			// TODO(@xiaoweim): Log ressource UID in the future if possible
+			r.Recorder.Event(&inputcr, "Normal", fmt.Sprintf("Finished expander stage %d: %s", index, expander), fmt.Sprintf("version: %d, resource count: %d", inputcr.GetGeneration(), applier.Count()))
+			for i, resourceStatus := range newStatus.Stages[expander].LastApplied {
+				logger.Info("Resource %d\n", i)
+				logger.Info("Resource name: %s, resource namespace: %s, resource gvk: %s %s %s, resource status: %s",
```

Comment:
I would consider having a method which returned an identifier string and call it at the start of each expander debug log line. You then know you are printing the same identifying log id everywhere. You even know the same grep will work for all of them.